addons/l10n_mx/controllers/portal.py

```diff
@@ -0,0 +1,14 @@
+from odoo.addons.portal.controllers import portal
+from odoo.http import request
+
+class CustomerPortal(portal.CustomerPortal):
+
+    def _get_mandatory_fields(self):
+        if request.env['res.country'].sudo().browse(int(request.params['country_id'])).code == 'MX':
```

Comment:
I wouldn't assume it will be a valid integer, after all this is used for validation in the first place.